client/engine/messageservice/test-messageservice.go

```diff
@@ -20,8 +22,9 @@ type TestMessageService struct {
 	address types.Address
 
 	// connection to Engine:
-	in  chan protocols.Message // for recieving messages from engine
-	out chan protocols.Message // for sending message to engine
+	in        chan protocols.Message // for recieving messages from engine
+	out       chan protocols.Message // for sending message to engine
+	meanDelay time.Duration          // average delay for messages
```

Comment:
Seeing as you are only letting the calling code specify the mean delay, it might be nice to at least document the distribution of delays that will be used. I think what you end up with is a uniform distribution over `[meanDelay/2, 3*meanDelay/2]`.